### PR TITLE
attempting to improve the api

### PR DESCRIPTION
**Pull request by tsloughter**
This is a bit of a mess still but I want to get it open for feed back.

Basically trying to simplify the span api, which has been moved to `oc_trace` and `ocp`. The `ocp` pdict version now only tracks a single span instead of the stack. This just feels cleaner... I'll add a macro for this as well.

As for `ctx`, I think it could be thought of as an alternative to the pdict. Both can hold the current span and tags.

We should also consider ctx simply carrying a pid or reference to entry in an ets table. I'm waiting to understand what information we need to be able to report about active spans before moving forward with that.

### CODE REVIEW

**Comment by deadtrickster** [comment]
do you have performance comparison?

**Comment by tsloughter** [comment]
@deadtrickster performance of what? Nothing on interacting with a span really changed.

In slack I'm thinking more about it and believe a good way forward is to provide `ocp` and a tracer process pool that tracks span data in an ets table.

But, importantly, we still expose all span manipulation functionality as is available today. And this requirement is actually in the census spec. So users will be able to pass around `Span4` after manipulating the data in `Span3`, `Span2`, `Span1` all they want :). Still a question of if we need to track those active spans or not, hopefully not.

**Comment by tsloughter** [comment]
For now the finished spans are simply added to the reporter and after reporting they are gone forever.

The next step will be adding the buckets to store spans after they finish separated by latencies by name. But that can come in a later PR.

**Comment by tsloughter** [comment]
@deadtrickster I think this PR is ready now.

File: src/oc_trace.erl [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+         annotation/2,
+
+         message_event/4,
+
+         set_status/3]).
+
+-include("opencensus.hrl").
+
+%% sampling bit is the first bit in 8-bit trace options
+-define(IS_ENABLED(X), (X band 1) =/= 0).
```
[the remark this answers is not in the document]
Comment (tsloughter) [reply src/oc_trace.erl]: Yes, it is exported now as `oc_trace:is_enabled/1`.

**Pull request merged**